spec/requests/api/orders_spec.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require 'swagger_helper'
+
+describe 'api/orders', type: :request do
+  path '/api/orders' do
+    get('list orders') do
+      tags 'Orders'
+      # type should be replaced with swagger 3.01 valid schema: {type: string} when rswag #317 is resoved:
+      # https://github.com/rswag/rswag/pull/319
+      parameter name: 'X-Spree-Token', in: :header, type: :string
+      parameter name: 'q[distributor_id_eq]', in: :query, type: :string, required: false, description: "Query orders for a specific distributor id."
+      parameter name: 'q[completed_at_gt]', in: :query, type: :string, required: false, description: "Query orders completed after a date."
+      parameter name: 'q[completed_at_lt]', in: :query, type: :string, required: false, description: "Query orders completed before a date."
+      parameter name: 'q[state_eq]', in: :query, type: :string, required: false, description: "Query orders by order state, eg 'cart', 'complete'."
+      parameter name: 'q[payment_state_eq]', in: :query, type: :string, required: false, description: "Query orders by order payment_state, eg 'balance_due', 'paid', 'failed'."
+      parameter name: 'q[email_cont]', in: :query, type: :string, required: false, description: "Query orders where the order email contains a string."
+      parameter name: 'q[order_cycle_id_eq]', in: :query, type: :string, required: false, description: "Query orders for a specific order_cycle id."
+      
+      response(200, 'get orders') do
+        # Adds model metadata for Swagger UI. Ideally we'd be able to just add:
+        # schema '$ref' => '#/components/schemas/Order_Concise'
+        # Which would also validate the response in the test, this is an open 
+        # issue with rswag: https://github.com/rswag/rswag/issues/268
+        metadata[:response][:content] = { "application/json": {
+            schema: {'$ref' => '#/components/schemas/Order_Concise'}
+          }
+        }
+        context "when there are four orders with different properties set" do
+          let(:order_dist_1) { create(:order_with_distributor, email: "specific_name@example.com") }
+          let(:order_dist_2) { create(:order_with_distributor) }
+          let(:order_dist_1_complete) { create(:order,  distributor: order_dist_1.distributor, state: 'complete', completed_at: Time.zone.today - 7.days) }
+          let(:order_dist_1_credit_owed) { create(:order,  distributor: order_dist_1.distributor, state: 'complete', payment_state: 'credit_owed', completed_at: Time.zone.today) }
+          
```

Comment:
I gave this a try, you need to use let! to get the orders created.

You can debug SearchOrders#fetch_orders for details.
This (using let!) gets 3 orders into the first test:
![image](https://user-images.githubusercontent.com/1640378/84902719-16648400-b0a5-11ea-9112-797043b74d76.png)


The 4th order is not showing up because the user has no permissions to edit  order_dist_2. If you want to get the 4 orders in the first test you need to make the user a manager or owner of order_dist_2.distributor

Reply:
with this extra setup the first test passes:
before { order_dist_2.distributor.update_attributes owner: user }

Reply:
Thanks - I knew it was gonna be something stupidly obvious. In my travels I did see a section on difference between let and let! but didn't read it

Reply:
:+1:

Reply:
Did I say I had zero experience with ruby and rails when I started in OFN 2 years ago? I mean, I have been there!
Code Review is very valuable, my advise is: dont spend too much time when you are stuck, get it out in a PR and someone with more rails on their hands will fix it for you. This was a good example :+1: 
The debugging part is important, get the byebug instruction in the right place and when there, see what's going on, in this case, I run editable_orders in SearchOrders#fetch_orders and then, because the orders where not showing there I run Spree::Orders.all and saw the orders were not created.

Reply:
Really, I'd never have guessed you were a ruby newbie so recently @luisramos0  - that's good to know!

I was thinking also I should have just checked the test DB whilst debugging the test - something I'm much more familiar with.

I'm definitely happy to share early and often. I am way past caring about looking stupid after this many years in dev ;-)